src/tribler/core/components/gigachannel_manager/tests/test_gigachannel_manager.py

```diff
@@ -222,6 +224,65 @@ def mock_process_channel_dir(c, _):
     assert not gigachannel_manager.channels_processing_queue
 
 
+async def test_check_channel_updates_for_different_states(gigachannel_manager, metadata_store):
+    with db_session:
+        def random_subscribed_channel():
+            return metadata_store.ChannelMetadata(
+                title=f"Channel {random.randint(0, 100)}",
+                public_key=os.urandom(32),
+                signature=os.urandom(32),
+                skip_key_check=True,
+                timestamp=123,
+                local_version=122,
+                subscribed=True,
+                infohash=random_infohash(),
+            )
+
+        # Three channels in different states based on the setup
+        channel_with_metainfo = random_subscribed_channel()
+        already_downloaded_channel = random_subscribed_channel()
+        non_downloaded_channel = random_subscribed_channel()
+
+        # Setup 1: metainfo is already available for channel torrent.
+        def mock_get_metainfo(infohash):
+            return MagicMock() if infohash == channel_with_metainfo.infohash else None
+
+        gigachannel_manager.download_manager.metainfo_requests = MagicMock()
+        gigachannel_manager.download_manager.metainfo_requests.get = mock_get_metainfo
```

Comment:
NIT: can be simpified
```suggestion
        gigachannel_manager.download_manager.metainfo_requests = MagicMock(get=mock_get_metainfo)
```